model/histogram/float_histogram.go

```diff
@@ -957,3 +957,73 @@ func (i *allFloatBucketIterator) Next() bool {
 func (i *allFloatBucketIterator) At() Bucket[float64] {
 	return i.currBucket
 }
+
+// AllReverseBucketIterator returns a BucketIterator to iterate over all negative,
+// zero, and positive buckets in descending order (starting at the lowest bucket
+// and going up). If the highest negative bucket or the lowest positive bucket
+// overlap with the zero bucket, their upper or lower boundary, respectively, is
+// set to the zero threshold.
+func (h *FloatHistogram) AllReverseBucketIterator() BucketIterator[float64] {
+	return &allReverseFloatBucketIterator{
+		h:       h,
+		negIter: h.NegativeBucketIterator(),
+		posIter: h.PositiveReverseBucketIterator(),
+		state:   1,
+	}
+}
+
+type allReverseFloatBucketIterator struct {
```

Comment:
`allReverseFloatBucketIterator` and its methods are almost the same as `allFloatBucketIterator` and its methods. I believe with some proper renaming (e.g. `leftIter` and `rightIter` rather than `negIter` and `posIter`), we can unify both and avoid the code duplication.